src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.NestedTypes.cs

```diff
@@ -102,12 +102,12 @@ public override bool Equals(object obj)
                     return false;
                 }
 
-                return ProviderId == other.ProviderId && StateTypeId == other.StateTypeId && Key == other.Key;
+                return Analyzer == other.Analyzer && StateTypeId == other.StateTypeId && Key == other.Key;
             }
 
             public override int GetHashCode()
             {
-                return Hash.Combine(Key, Hash.Combine(ProviderId, (int)StateTypeId));
+                return Hash.Combine(Key, Hash.Combine(Analyzer.GetHashCode(), (int)StateTypeId));
```

Comment:
Now that I think about it, any of our caches/conditional weak tables based on DiagnosticAnalyzer as key can do so as well. Should we seal Equals and GetHashCode in DiagnosticAnalyzer type?
@JohnHamby @srivatsn thoughts?